src/lib/classes/QContext.svelte.ts

```diff
@@ -0,0 +1,27 @@
+import { getContext, setContext } from "svelte";
+
+export class QContext<T> {
+  #state = $state<T>();
```

Comment:
There is a difference between `private`, `protected` and `#` though.
`private` => makes the state invisible by the type checker (a user can still access `state` it if they ignore the TS error)
`protected` => idem but rather than making `state` visible for the class only, it makes it usable for classes extending it too
`#` => it's a JS feature that prevents users being able to access the property outside the class

Now, `QContext` is, unless we make it public API, a private API so using `private` or `protected` should not cause problems. I let you decide which to use as I don't have a strong opinion.
Just know that if we expose is to the public, using `#` is better.